unix_integration/common/src/unix_passwd.rs

```diff
@@ -27,36 +25,50 @@ pub fn parse_etc_passwd(bytes: &[u8]) -> Result<Vec<EtcUser>, UnixIntegrationErr
         .collect::<Result<Vec<EtcUser>, UnixIntegrationError>>()
 }
 
-fn members<'de, D>(deserializer: D) -> Result<Vec<String>, D::Error>
-where
-    D: Deserializer<'de>,
-{
-    struct InnerCsv;
-
-    impl<'de> Visitor<'de> for InnerCsv {
-        type Value = Vec<String>;
-
-        fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
-            formatter.write_str("string")
-        }
-
-        fn visit_str<E>(self, value: &str) -> Result<Vec<String>, E>
-        where
-            E: de::Error,
-        {
-            Ok(value.split(',').map(|s| s.to_string()).collect())
-        }
-    }
+#[serde_as]
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+pub struct EtcShadow {
+    pub name: String,
+    pub password: String,
+    // 0 means must change next login.
+    // None means all other aging features are disabled
+    pub epoch_change_days: Option<i64>,
+    // 0 means no age
+    #[serde_as(deserialize_as = "DefaultOnNull")]
+    pub days_min_password_age: i64,
+    pub days_max_password_age: Option<i64>,
+    // 0 means no warning
+    #[serde_as(deserialize_as = "DefaultOnNull")]
+    pub days_warning_period: i64,
+    // Number of days after max_password_age passes where the password can
+    // still be accepted such that the user can update their password
+    pub days_inactivity_period: Option<i64>,
+    pub epoch_expire_date: Option<i64>,
+    pub flag_reserved: Option<u32>,
+}
 
-    deserializer.deserialize_str(InnerCsv)
+pub fn parse_etc_shadow(bytes: &[u8]) -> Result<Vec<EtcShadow>, UnixIntegrationError> {
+    use csv::ReaderBuilder;
+    let mut rdr = ReaderBuilder::new()
+        .has_headers(false)
+        .delimiter(b':')
+        .from_reader(bytes);
+    rdr.deserialize()
+        .map(|result| {
+            result
+                .inspect_err(|err| eprintln!("{:?}", err))
+                .map_err(|_e| UnixIntegrationError)
```

Comment:
Shouldn't this just be a map_err ?
```suggestion
                .map_err(|err| {
                  eprintln!("{:?}", err));
                  UnixIntegrationError
                  })
```

Reply:
Clippy auto fix seems to have not autofixed lol.

Reply:
Bad clippy, no donut. Probably because you're not just returning something-err